src/CodeGenHelpers/DocumentationComment.cs

```diff
@@ -0,0 +1,37 @@
+using System.Collections.Generic;
+
+namespace CodeGenHelpers
+{
+    public class DocumentationComment
+    {
+        public string? Summary { get; set; }
+
+        public Dictionary<string, string> ParameterDoc { get; } = new Dictionary<string, string>();
+
+        public bool InheritDoc { get; set; }
+
+        public string? InheritFrom { get; set; }
+
+        internal void Write(ref CodeWriter writer)
+        {
+            if (InheritDoc)
+            {
+                writer.AppendLine($"/// <inheritdoc {(InheritFrom is null ? "" : $"cref=\"{InheritFrom}\"")}/>");
+                return;
+            }
+
+            if (Summary is {})
+            {
+                writer.AppendLine("/// <summary>");
+
+                foreach (string line in Summary.Split('\n'))
```

Comment:
if we're splitting on \n then we probably want to remove \r, or add it as a split on option